Add get_all_sysmeta function

drkrynstrng opened the pull request:
I created a function to get the system metadata for all objects in a data package and return them as a list. The intended use is to make it easier to inspect the system metadata for every object in a package using one line of code. This commit is the basic function, but It could be enhanced to provide more warnings, return a message that the package is a parent package, etc. I organized it with the helper functions and have not exported it.

Also, the newer version of `roxygen2` seems to have modified the formatting for some of the documentation files after running `devtools::document()`, which is why they are showing up as modified in this commit.

dmullen17 commented on R/helpers.R:
```diff
@@ -298,3 +301,50 @@ create_dummy_enumeratedDomain_dataframe <- function(factors) {
+#' # Create separate object
+#' sysmeta_md <- all[["Metadata"]]
+#' }
+get_all_sysmeta <- function(node, pid) {
```
Could this be improved by adding an optional `child_packages = FALSE` argument?  In some cases there could be so many children you'd want to omit them, but if you're truly getting all the sysmeta from a package you might want to include these if they exist.

drkrynstrng replied:
Yeah, I think that would be a good option to have. I'll see if I can add that in easily.

dmullen17 commented on R/helpers.R:
```diff
@@ -298,3 +301,50 @@ create_dummy_enumeratedDomain_dataframe <- function(factors) {
+
+  pkg <- get_package(node, pid, file_names = TRUE)
+
+  all <- lapply(pkg$data, function(x) {dataone::getSystemMetadata(node, x)})
```
Irene's solr guide is also good to review https://nceas.github.io/datateam-training/reference/solr-queries.html

drkrynstrng replied:
Didn't know that, but I can see why it could be problematic. I'll modify the function to use a solr query instead.

drkrynstrng commented:
I've changed the code to use a Solr query, added an argument for child packages, and also added some tests.

Please check the translation from Solr to SystemMetadata. As far as I know, there is no corresponding Solr field for `sysmeta@serialVersion` and `sysmeta@archived`. I left these fields at their defaults. For `sysmeta@serialVersion` the default is 1, although I almost always see it as 0. For `sysmeta@archived`, the default is FALSE. Please advise if these defaults are problematic.

Additionally, the Solr fields for `sysmeta@dateUploaded` and `sysmeta@dateSysMetadataModified` have slightly different date/time formats than the corresponding SystemMetadata fields. These fields are ultimately treated as character in SystemMetadata, so I'm guessing it's okay.

jeanetteclark commented:
@dmullen17 or @amoeba (only if you have time) can you review Derek's most recent check?

drkrynstrng commented:
I discovered the `validate()` function in the datapack package, and it confirms that `get_all_sysmeta()` returns valid SystemMetadata objects.

jeanetteclark commented:
Hey @drkrynstrng this LGTM I can merge once you resolve the merge conflict
